### PR TITLE
Better testcase failure descriptions and command matching

### PR DESCRIPTION
**Pull request by dcbw**
@danwinship @alexanderConstantinescu @JacobTanenbaum @pecameron

### CODE REVIEW

File: go-controller/pkg/ovn/policy_old_test.go [hunk 1]
```diff
@@ -251,13 +251,6 @@ func (n networkPolicyOld) delPodCmds(fexec *ovntest.FakeExec, pod pod, networkPo
 	fexec.AddFakeCmdsNoOutputNoError([]string{
 		fmt.Sprintf("ovn-nbctl --timeout=15 clear address_set %s addresses", hashedNamespace),
 	})
-	fexec.AddFakeCmd(&ovntest.ExpectedCmd{
```
Comment (dcbw) [inline go-controller/pkg/ovn/policy_old_test.go]: Note that this is a drive-by fix to policy_old.go because it doesn't use port groups, but multicast depends on port groups, so we shouldn't be setting multicast enabled in policy_old.go at all.

File: go-controller/pkg/ovn/policy_test.go [hunk 2]
```diff
@@ -542,7 +548,8 @@ var _ = Describe("OVN NetworkPolicy Operations", func() {
 				nTest.baseCmds(fExec, namespace1, namespace2)
 				nTest.addCmds(fExec, namespace2)
 				nTest.addCmdsWithPods(fExec, nPodTest, namespace1)
-				npTest.addPodSelectorCmds(fExec, nPodTest, networkPolicy, true, true)
+				npTest.addNamespaceSelectorCmds(fExec, networkPolicy, true)
```
Comment (dcbw) [inline go-controller/pkg/ovn/policy_test.go]: This test looked wrong; addPodSelectorCmds() is supposed to be used for NPs that have a `podSelector` which this test is not exercising (it only has a `namespaceSelector`).

**Comment by squeed** [comment]
Gave it a quick scan, makes sense.

**Comment by alexanderConstantinescu** [comment]
Thanks for this PR! Hopefully it will ease our lives debugging and writing tests

File: go-controller/pkg/testing/testing.go [hunk 3]
```diff
@@ -46,44 +47,73 @@ func NewLooseCompareFakeExec() *FakeExec {
...
+		expectedCommands: make([]*ExpectedCmd, 0),
 	}
 }
 
+const fakeBinPrefix string = "/fake-bin/"
+
 // LookPath is for finding the path of a file
 func (f *FakeExec) LookPath(file string) (string, error) {
-	return f.LookPathFunc(file)
+	return fakeBinPrefix + file, nil
```
[the remark this answers is not in the document]
Comment (dcbw) [reply go-controller/pkg/testing/testing.go]: @alexanderConstantinescu LookPath() is an interface though, so i can't clean that up even if we don't use it...
